gradle-jdks/src/main/java/com/palantir/gradle/jdks/JdksExtension.java

```diff
@@ -57,6 +63,10 @@ public JdksExtension() {
         this.jdkStorageLocation = SynchronizedInterface.synchronizeAllInterfaceMethods(
                 DirectoryProperty.class, getObjectFactory().directoryProperty());
         this.daemonTarget = getObjectFactory().property(JavaLanguageVersion.class);
+        this.jdkMajorVersionsToUse = getObjectFactory().setProperty(JavaLanguageVersion.class);
+        this.jdkMajorVersionsToUse.convention(Arrays.stream(JavaVersion.values())
+                .map(javaVersion -> JavaLanguageVersion.of(javaVersion.getMajorVersion()))
+                .collect(Collectors.toSet()));
```

Comment:
atm, we are generating files for all the jdks that are configured in the jdks extension (11, 17, 21, ...) regardless if they are used or not. When `./gradlew` script runs, it will install all the jdks for which we generated files. 

In the next PR (wip PR), I am going to set `jdkMajorVersionsToUse ` to the jdk versions we actually use (based on `baseline-java-versions` configurations)